File: MoreModulePractice/app/src/main/java/com/bdpqchen/moremodulepractice/MainActivity.java
package com.bdpqchen.moremodulepractice;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.bdpqchen.modulea.HomeActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tv)
    TextView tv;
    @BindView(R.id.button)
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        View view = MainActivity.class;
        ButterKnife.bind(this);

    }

    @OnClick({R.id.tv, R.id.button})
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.tv:
                startActivity(new Intent(this, HomeActivity.class));

                break;
            case R.id.button:
                startActivity(new Intent(this, HomeActivity.class));

                break;
        }
    }
}

File: MagicImage/app/src/main/java/com/bdpqchen/magicimage/MainActivity.kt
package com.bdpqchen.magicimage

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import com.bdpqchen.magicimage.old.FilterActivity
import com.bdpqchen.magicimage.old.PanoramaActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val panorama = findViewById<Button>(R.id.panorama)
        panorama.setOnClickListener({ startAction(PanoramaActivity::class.java) })
        val filter = findViewById<Button>(R.id.filter)
        filter.setOnClickListener({ startAction(HomeActivity::class.java) })
        val wave = findViewById<Button>(R.id.wave)
        wave.setOnClickListener({ startAction(WaveActivity::class.java) })
        findViewById<Button>(R.id.basicColor).setOnClickListener({
            startAction(BasicColorActivity::class.java)
        })
        findViewById<Button>(R.id.matrixCalculate).setOnClickListener({
            startAction(MatrixActivity::class.java)
        })
    }

    private fun startAction(clazz: Class<*>) {
        startActivity(Intent(this, clazz))
    }


}
